libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp

```diff
@@ -235,53 +230,90 @@ void AP_RCProtocol_CRSF::_process_byte(uint32_t timestamp_us, uint8_t byte)
 
     // need a header to get the length
     if (_frame_ofs < CSRF_HEADER_TYPE_LEN) {
-        return;
+        return true;
     }
 
     if (_frame.device_address != DeviceAddress::CRSF_ADDRESS_FLIGHT_CONTROLLER) {
-        return;
+        return false;
     }
 
     // parse the length
     if (_frame_ofs == CSRF_HEADER_TYPE_LEN) {
         _frame_crc = crc8_dvb_s2(0, _frame.type);
         // check for garbage frame
         if (_frame.length > CRSF_FRAME_PAYLOAD_MAX) {
-            _frame_ofs = 0;
+            return false;
         }
-        return;
+        return true;
     }
 
     // update crc
     if (_frame_ofs < _frame.length + CSRF_HEADER_LEN) {
         _frame_crc = crc8_dvb_s2(_frame_crc, byte);
     }
 
-    // overflow check
+    // overflow check, should never happen
     if (_frame_ofs > _frame.length + CSRF_HEADER_LEN) {
-        _frame_ofs = 0;
-        return;
+        return false;
     }
 
     // decode whatever we got and expect
     if (_frame_ofs == _frame.length + CSRF_HEADER_LEN) {
-        log_data(AP_RCProtocol::CRSF, timestamp_us, (const uint8_t*)&_frame, _frame_ofs - CSRF_HEADER_LEN);
-
-        // we consumed the partial frame, reset
-        _frame_ofs = 0;
-
         // bad CRC (payload start is +1 from frame start, so need to subtract that from frame length to get index)
         if (_frame_crc != _frame.payload[_frame.length - 2]) {
-            return;
+            return false;
         }
 
+        log_data(AP_RCProtocol::CRSF, timestamp_us, (const uint8_t*)&_frame, _frame_ofs - CSRF_HEADER_LEN);
+
+        // we consumed the partial frame, reset
+        _frame_ofs = 0;
         _last_frame_time_us = _last_rx_frame_time_us = timestamp_us;
+    
         // decode here
         if (decode_crsf_packet()) {
             _last_tx_frame_time_us = timestamp_us;  // we have received a frame from the transmitter
             add_input(MAX_CHANNELS, _channels, false, _link_status.rssi, _link_status.link_quality);
         }
+    } else if (_frame_ofs >= CRSF_FRAMELEN_MAX) {   // overflow check
+        return false;
+    }
+
+
+    return true;
+}
+
+bool AP_RCProtocol_CRSF::skip_to_next_frame(uint32_t timestamp_us)
+{
+    // need to check from after the current device address marker (0xC8)
+    uint8_t* frame_start = (uint8_t*)memchr(&_frame + sizeof(uint8_t), DeviceAddress::CRSF_ADDRESS_FLIGHT_CONTROLLER, _frame_ofs - 1);
```

Comment:
the maths here is quite complex, can we get some captures we could run through a SITL CRSF decoder to test this with some corrupt data?

Reply:
Done